packages/web/app/forms/MedicationForm.jsx

```diff
@@ -497,6 +498,12 @@ export const MedicationForm = ({ encounterId, onCancel, onSaved }) => {
     setSubmittedMedication(newMedication);
   };
 
+  const onFinalise = async ({ data, isPrinting, submitForm }) => {
+    setAwaitingPrint(isPrinting);
+    await submitForm(data);
+    await queryClient.invalidateQueries(['encounterMedication', encounterId]);
```

Comment:
This is an optional suggestion but invalidateQueries could go in the onSuccess callback. It might catch an unexpected edge case or make it less likely someone misses onFinalise in future